### DISCUSSION TITLE
Example CMake project

### DISCUSSION THREAD

**Discussion by CsatiZoltan** [DLR-AMR/t8code#786] [Setup & Installation]
Hi,

If any of you have a CMake-based C++ application that uses t8code, could you please share the CMakeLists.txt file?

**Comment by CsatiZoltan** [comment]
I managed to set up a minimal example. I will insert the CMakeFile here and edit it from time to time. If you have ideas to make it better, let us discuss it. The idea is to create a robust solution that will hopefully help others as well.

This is a CMake project with a single source file *main.cpp*, using t8code compiled with VTK and MPI support.
It is assumed that the environment variables *T8_INCLUDE* and *T8_LIB* are set to the path of the t8code header files and libraries.
I was inspired by the examples provided in [this comment](https://github.com/orgs/DLR-AMR/discussions/786#discussioncomment-7207070).

```cmake
cmake_minimum_required(VERSION 3.12.0)
project(project VERSION 0.1.0 LANGUAGES C CXX)

# If t8code was build with MPI support and linked with VTK
find_package(MPI REQUIRED)
find_package(VTK REQUIRED)

include_directories($ENV{T8_INCLUDE})
link_directories($ENV{T8_LIB})

add_executable(project main.cpp)

target_link_libraries(project MPI::MPI_C)
target_link_libraries(project sc p4est t8)
```

Remarks:
1. CMake recommends lower case command names, e.g. `include_directories` instead of `INCLUDE_DIRECTORIES`.
2. I increased the minimum CMake version number to be able to use `find_package(VTK)`.
3. If you want to use this CMakeFile from Visual Studio Code, install the [CMake Tools](https://github.com/microsoft/vscode-cmake-tools) extension, and update your *settings.json* file with the following setting:
    ```
    "cmake.environment": {
        "VTK_DIR": "<path_to_vtk-config.cmake>",
        "T8_INCLUDE": "<same_path_as_defined_in_the_environment_variable_above>",
        "T8_LIB": "<same_path_as_defined_in_the_environment_variable_above>"
    }
    ```
    where *VTK_DIR* needs to be given only if `find_package(VTK)` fails.

**Comment by jmark** [comment]
https://github.com/DLR-AMR/png2mesh/blob/main/CMakeLists.txt
https://github.com/DLR-AMR/mesh2tikz/blob/main/CMakeLists.txt
